Includes/findXBE.cpp

```diff
@@ -4,13 +4,14 @@ int findXBE(void* list) {
   xbeFinderArg *itm = static_cast<xbeFinderArg*>(list);
   vector<xbeMenuItem>* gmi_list = itm->list;
   char* path = const_cast<char*>(itm->path);//("C:\\");
-  char* mask = const_cast<char*>("*");
 #ifdef NXDK
-  int ret;
   char tmp[64];
+  char xbeName[XBENAMESIZE];
```

Comment:
What if all 40 of the `XBENAMESIZE` letters are being used, and the original field had no zero termination? Can this happen? I think this should be `XBENAMESIZE + 1` and you should force a zero termination after the loop.